pkg/project/apiserver/admission/nodeenv/admission.go

```diff
@@ -60,6 +64,15 @@ func (p *podNodeEnvironment) admit(a admission.Attributes, mutationAllowed bool)
 	if err != nil {
 		return apierrors.NewForbidden(resource, name, err)
 	}
+
+	// If scheduler.alpha.kubernetes.io/node-selector is set on the pod,
+	// do not process the pod further.
+	if len(namespace.ObjectMeta.Annotations) > 0 {
```

Comment:
Is there any way this could break an existing cluster?

Reply:
Yes i think, in case both nodeenv and podnodeselector are being used and that means both `openshift.io/node-selector` and `scheduler.alpha.kubernetes.io/node-selector` are configured. For example, lets say pod has k1=v1, and `openshift.io/node-selector` is `k2=v2` and `scheduler.alpha.kubernetes.io/node-selector` is `k3=v3` so result would be all three (`k1=v1`,`k2=v2`,`k3=v3`), but with this PR result would be (`k1=v1`, `k3=v3`).  Though I am not sure how frequently both  nodeenv and podnodeselector are being used together. 

As far as I remember, the initial design thoughts were to have possibility of processing both upstream  (`scheduler.alpha.kubernetes.io/node-selector`) and downstream (`openshift.io/node-selector`) at the same time. So this PR breaks that. But dont see ant choice so far with conformance tests.

Also I noticed that I need to modify vendor/k8s.io/kubernetes/pkg/controller/daemon/patch_nodeselector.go too to make DS controller in sync with this change, because patched DS controller also takes the same approach i.e. processing both node selectors annotations at the same time. 

So I am still testing my changes to vendor/k8s.io/kubernetes/pkg/controller/daemon/patch_nodeselector.go.